kernel/src/fs/procfs/stat.rs

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! This module offers `/proc/stat` file support, which provides
+//! information about kernel system statistics.
+//!
+//! Reference: <https://man7.org/linux/man-pages/man5/proc_stat.5.html>
+
+use alloc::format;
+
+use aster_softirq::{
+    collect_per_irq_counts_across_all_cpus, collect_per_softirq_counts_across_all_cpus,
+    softirq_id::*,
+};
+
+use crate::{
+    fs::{
+        procfs::template::{FileOps, ProcFileBuilder},
+        utils::Inode,
+    },
+    prelude::*,
+    process::forks_count,
+    sched::nr_queued_and_running,
+    thread::context_switch_count,
+    time::{cpu_time_stats::CpuTimeStatsManager, SystemTime, START_TIME},
+};
+
+pub struct StatFileOps;
+
+impl StatFileOps {
+    pub fn new_inode(parent: Weak<dyn Inode>) -> Arc<dyn Inode> {
+        ProcFileBuilder::new(Self).parent(parent).build().unwrap()
+    }
+
+    fn collect_stats() -> String {
+        let cpu_manager = CpuTimeStatsManager::singleton();
+
+        // Get global CPU statistics
+        let global_stats = cpu_manager.collect_stats_on_all_cpus();
+
+        let mut output = String::new();
+
+        // Global CPU line: cpu <user> <nice> <system> <idle> <iowait> <irq> <softirq> <steal> <guest> <guest_nice>
+        output.push_str(&format!(
```

Comment:
Use `write!`/`writeln!` to generate procfs output, like this:
https://github.com/asterinas/asterinas/blob/d4d84f18b7288f7251c779aad5749de9ffbe368f/kernel/src/fs/procfs/pid/task/status.rs#L98-L100

This pattern will enable us to avoid heap allocation, e.g., if we have implemented something similar to https://github.com/asterinas/asterinas/pull/2160#discussion_r2322555452.